### PR TITLE
api: Update type annotation in 'remove_subscriptions'.

### PR DESCRIPTION
**Pull request by akashaviator**
This updates the type annotation for the 'principals' argument
to accept list of email addresses as well as user ids.

### CODE REVIEW

File: zulip/zulip/__init__.py [hunk 1]
```diff
@@ -1270,7 +1270,8 @@ def add_subscriptions(self, streams: Iterable[Dict[str, Any]], **kwargs: Any) ->
             request=request,
         )
 
-    def remove_subscriptions(self, streams: Iterable[str], principals: Optional[Iterable[str]] = None) -> Dict[str, Any]:
+    def remove_subscriptions(self, streams: Iterable[str],
+                             principals: Optional[List[Any]] = None) -> Dict[str, Any]:
```
Comment (timabbott) [inline zulip/zulip/__init__.py:1274]: This should be typed `Union[List[str], List[int]]`.

**Comment by akashaviator** [comment]
@timabbott I've updated the PR.

File: zulip/zulip/__init__.py [hunk 2]
```diff
@@ -1270,7 +1270,8 @@ def add_subscriptions(self, streams: Iterable[Dict[str, Any]], **kwargs: Any) ->
             request=request,
         )
 
-    def remove_subscriptions(self, streams: Iterable[str], principals: Optional[Iterable[str]] = None) -> Dict[str, Any]:
+    def remove_subscriptions(self, streams: Iterable[str],
+                             principals: Optional[List[Union[str, int]]] = None) -> Dict[str, Any]:
```
Comment (timabbott) [inline zulip/zulip/__init__.py:1274]: It should be a `Union[List`, not `List[Union`, since all the things in a list must be the same type.

**Comment by akashaviator** [comment]
@timabbott Can you have a look ?

File: zulip/zulip/__init__.py [hunk 3]
```diff
@@ -1270,16 +1270,15 @@ def add_subscriptions(self, streams: Iterable[Dict[str, Any]], **kwargs: Any) ->
...
         )
 
-    def remove_subscriptions(self, streams: Iterable[str], principals: Optional[Iterable[str]] = None) -> Dict[str, Any]:
+    def remove_subscriptions(self, streams: Iterable[str],
+                             principals: Optional[Union[List[str], List[int]]] = None) -> Dict[str, Any]:
         '''
             See examples/unsubscribe for example usage.
         '''
-        if principals is None:
-            principals = []
```
Comment (timabbott) [inline zulip/zulip/__init__.py:1278]: Why did you change this?  I think it was better with this on its own line.

Comment (showell) [reply zulip/zulip/__init__.py:1278]: @timabbott The reason the change happened here was to work around `mypy` being finicky about shadowing a variable and thinking we've changed the type of it to `List[Any]` (figuratively).

More discussion here: https://chat.zulip.org/#narrow/stream/92-learning/topic/mypy.20error/near/902548

Are we enforcing that the Python versions >= 3.6 yet for the API?  This would influence our decision here on how best to deal with mypy.

Comment (akashaviator) [reply zulip/zulip/__init__.py:1278]: @timabbott I was facing an issue. I've explained it here: https://chat.zulip.org/#narrow/stream/92-learning/topic/mypy.20error

Comment (timabbott) [reply zulip/zulip/__init__.py:1278]: We probably shouldn't limit this to disallow 3.5 before it reaches EOL.  Seems really cheap to keep support and I'm sure someone will have a bad experience if we do.

**Comment by akashaviator** [comment]
@timabbott Can you have a look ?

**Comment by timabbott** [comment]
I don't think this latest change is an improvement over what we have in master.

**Comment by akashaviator** [comment]
@timabbott Can you have a look ?

**Comment by timabbott** [comment]
Merged, thanks @akashaviator!  `Sequence` is the right way to do this.
